vc-worker/src/main/java/com/vmware/flowgate/vcworker/scheduler/job/VCDataService.java

```diff
@@ -168,11 +189,489 @@ public void checkAndUpdateIntegrationStatus(SDDCSoftwareConfig vc, String messag
+            }
+
+            if (newHostFlag || hardwareSummary.getCpuMhz() != hostInfo.getSingleCoreCpuMhz()) {
+               hostInfo.setSingleCoreCpuMhz(hardwareSummary.getCpuMhz());
+               needUpdate = true;
+            }
+
+            if (newHostFlag || hardwareSummary.getMemorySize() != hostInfo.getMemoryCapacity()) {
+               hostInfo.setMemoryCapacity(hardwareSummary.getMemorySize());
+               needUpdate = true;
+            }
+         }
+      }
+
+      Map<String, HostNic> newHostNicMap = new HashMap<>();
+      Map<String, HostNic> oldHostNicMap = new HashMap<>();
+
+      NetworkInfo networkInfo = host.getConfig().getNetwork();
+      if (networkInfo != null) {
+         PhysicalNic[] physicalNics = networkInfo.getPnic();
+         int nicsNum = physicalNics.length;
+         if (nicsNum > 0) {
+            List<HostNic> hostNics = new ArrayList<HostNic>();
+
+            for (int i = 0; i < nicsNum; i++) {
+
+               HostNic hostNic = new HostNic();
+
+               hostNic.setMacAddress(physicalNics[i].getMac());
+               hostNic.setDriver(physicalNics[i].getDriver());
+               hostNic.setDuplex(physicalNics[i].getLinkSpeed() == null ? false
+                     : physicalNics[i].getLinkSpeed().isDuplex());
+               hostNic.setLinkSpeedMb(physicalNics[i].getLinkSpeed() == null ? 0
+                     : physicalNics[i].getLinkSpeed().getSpeedMb());
+               hostNic.setName(physicalNics[i].getDevice());
+               hostNics.add(hostNic);
+               newHostNicMap.put(physicalNics[i].getMac(), hostNic);
+            }
+            if (newHostFlag) {
+               hostInfo.setHostNics(hostNics);
+            } else {
+               List<HostNic> oldNics = hostInfo.getHostNics();
+               for (HostNic hostNic : oldNics) {
+                  oldHostNicMap.put(hostNic.getMacAddress(), hostNic);
+               }
+
+               if (newHostNicMap.size() != oldHostNicMap.size()) {
+
+                  for (Map.Entry<String, HostNic> entry : oldHostNicMap.entrySet()) {
+                     HostNic oldValue = entry.getValue();
+                     HostNic newValue = newHostNicMap.get(entry.getKey());
+                     if (!oldValue.equals(newValue)) {
+                        needUpdate = true;
+                        hostInfo.setHostNics(hostNics);
+                        break;
+                     }
+                  }
+               }
+            }
+         }
+      }
+
+      long diskCapacity = 0;
+      DatastoreInfo[] datastores = host.queryConnectionInfo().getDatastore();
+      if (datastores != null && datastores.length > 0) {
+         for (DatastoreInfo datastore : datastores) {
+            diskCapacity += datastore.getSummary().getCapacity();
+         }
+         if (newHostFlag || diskCapacity != hostInfo.getDiskCapacity()) {
+            needUpdate = true;
+            hostInfo.setDiskCapacity(diskCapacity);
```

Comment:
TODO:  only count writeable datastore.

Reply:
add comment. say we will count shared datastore for each host, which will cause problem in calculate the total available storage among multiple hosts.